ring/lifecycler_test.go

```diff
@@ -445,6 +445,49 @@ func TestLifecycler_ZonesCount(t *testing.T) {
 	}
 }
 
+func TestLifecycler_Zones(t *testing.T) {
```

Comment:
Feedback from Marco: Ensure we get more cases in the unit tests to test when instances are leaving, etc